distribution/errors_test.go

```diff
@@ -0,0 +1,56 @@
+package distribution
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/registry/api/errcode"
+	"github.com/docker/distribution/registry/api/v2"
+	"github.com/docker/distribution/registry/client"
+)
+
+func TestContinueOnError(t *testing.T) {
```

Comment:
While I am not sure if we have a better way of testing this, this kind of unit is just testing the implementation detail of the package. This kind of test makes it harder or impossible to refactor in the future.

Is there another way we could setup this test?

If this is the only way, it should at least be a table-based test.